### PR TITLE
Fix resetting user's usage value in update usage

### PR DESCRIPTION
**Pull request by Iam54r1n4**

### CODE REVIEW

**Comment by Iam54r1n4** [comment]
@hiddify-com Review please

File: hiddifypanel/models/user.py [hunk 1]
```diff
@@ -148,6 +148,18 @@ def user_should_reset(self) -> bool:
 
         return res
 
+    def reset(self, commit: bool = False):
+        '''Resets the user usages'''
+        self.last_reset_time = datetime.date.today()
+        self.current_usage_GB = 0
+
+        # there's no usage of UserDetail yet, but we reset it too
+        if ud := UserDetail.query.filter(UserDetail.user_id == self.id).first():
```
Comment (hiddify-com) [inline hiddifypanel/models/user.py]: UserDetail 
نباید ریست بشه
این تاریخچه هست

File: hiddifypanel/panel/usage.py [hunk 2]
```diff
@@ -42,66 +42,84 @@ def _add_users_usage(users_usage_data: Dict[User, Dict], child_id, sync=False):
             daily_usage[adm.id] = DailyUsage(admin_id=adm.id, child_id=child_id)
             db.session.add(daily_usage[adm.id])
         daily_usage[adm.id].online = User.query.filter(User.added_by == adm.id).filter(func.DATE(User.last_online) == today).count()
-    # db.session.commit()
+
+    # reset users usage if needed
+    # TODO: optimize the resetting operation
+    for user in User.query.all():
+        if user.user_should_reset():
+            user.reset(commit=False)
```
Comment (hiddify-com) [inline hiddifypanel/panel/usage.py]: rename it to reset_usage

File: hiddifypanel/panel/usage.py [hunk 3]
```diff
@@ -42,66 +42,84 @@ def _add_users_usage(users_usage_data: Dict[User, Dict], child_id, sync=False):
...
+
     userDetails = {p.user_id: p for p in UserDetail.query.filter(UserDetail.child_id == child_id).all()}
     for user, uinfo in users_usage_data.items():
         usage_bytes = uinfo['usage']
         devices = uinfo['devices']
-        # user_active_before=user.is_active
+
+        # UserDetails things
         detail = userDetails.get(user.id)
```
Comment (hiddify-com) [inline hiddifypanel/panel/usage.py]: user details 
دوتا کلید داره یکی id  یکی child id

File: hiddifypanel/panel/usage.py [hunk 4]
```diff
@@ -42,66 +42,84 @@ def _add_users_usage(users_usage_data: Dict[User, Dict], child_id, sync=False):
...
         if not detail:
             detail = UserDetail(user_id=user.id, child_id=child_id)
             detail.current_usage_GB = detail.current_usage_GB or 0
```
Comment (hiddify-com) [inline hiddifypanel/panel/usage.py]: اضافه هست

File: hiddifypanel/panel/usage.py (hunk 2 above)
Comment (hiddify-com) [reply hiddifypanel/panel/usage.py]: چرا توی دو تا for آوردی؟ اینجوری باعث میشه دوبار از دیتایس بخونه و سربار اضافه میشه

Comment (Iam54r1n4) [reply hiddifypanel/panel/usage.py]: The function has proper documentation that indicates what the function is doing, so there is no need to change the name of the function. Additionally, there isn't any similar function with this concept in the user model, so there won't be any conflict.

The users_usage_data doesn't necessarily have all the users. Actually, the problem lies in this itself. Consider that a user is disabled; it wouldn't exist in this variable. Therefore, the user's usage won't be reset.

**Pull request merged**